contracts/contracts/src/functioncall/gpact/CrosschainControl.sol

```diff
@@ -452,6 +454,8 @@ contract CrosschainControl is
                 );
                 activeCallFailed = true;
             }
+        } else {
```

Comment:
If the call failed, activeCallFailed will already be set. There is nothing else to do, as the Segment or Root event will record that the function call failed.

Reply:
Hence, this could should be removed.

Reply:
The failure of a segment should propagate up from one segment to another to the root.